src/modules/token/fungible/utils/lsk.js

```diff
@@ -41,6 +41,45 @@ export const toRawLsk = (value) => {
   return new BigNumber(amount * new BigNumber(10).pow(8)).decimalPlaces(0).toNumber();
 };
 
+const getTokenDecimals = (token) => {
+  const { decimals } = token.denomUnits?.find?.(({ denom }) => denom === token.displayDenom) || {};
+
+  return decimals;
+};
+
+/**
+ * Converts a given token amount to its token symbol denom
+ *
+ * @param {BigNumber|number} amount - Amount value to be converted
+ * @param {BigNumber|number} token - Token value merged with its equivalent metadata
+ * @returns {BigNumber} Amount value converted to the token symbol's denom
+ */
+export const convertFromBaseDenom = (amount, token = {}) => {
+  const decimals = getTokenDecimals(token);
+
+  if (!decimals) return '0';
+
+  return new BigNumber(amount || 0).dividedBy(new BigNumber(BASE).pow(decimals)).toFixed();
+};
+
+/**
+ * Converts from denom denoted by the token symobl to a smaller unit
+ *
+ * @param {BigNumber|number} amount - Amount value to be converted
+ * @param {BigNumber|number} token - Token value merged with its equivalent metadata
+ * @returns {BigNumber} Amount value converted from the token symbol's denom
+ */
+export const convertToBaseDenom = (amount, token = {}) => {
+  // eslint-disable-next-line no-restricted-globals
+  if (isNaN(amount)) return '0';
```

Comment:
@eniolam1000752 `Number.isNaN` should be used here

Reply:
`Number.isNaN` would not work here because `Number.isNaN` would return `false` even though `abc` is provided which in this use case its wrong (Number.isNaN tests that the provided parameter can be represented by an equivalent numeric value) and what I want to test is to ensure that the string value provided is a number and window.isNaN does that

https://stackoverflow.com/questions/25176459/is-number-isnan-more-broken-than-isnan